mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/parser/StringCombinators.kt

```diff
@@ -23,6 +25,30 @@ object StringCombinators {
   /** Parses the first digit of a [String], if it exists. */
   fun digit(): Parser<String, Int> = char().filter { it in '0'..'9' }.map { it - '0' }
 
+  /**
+   * Parses the first [Token] of a [String]
+   *
+   * @param delimiter the delimiter between each token
+   */
+  fun token(delimiter: Char = ' '): Parser<String, Token> = Parser {
+    if (it.isNotEmpty()) {
+      val splitString = it.trim { char -> char == delimiter }.split(delimiter, limit = 2)
+      val result = splitString.first()
+      val remaining = splitString.drop(1).firstOrNull() ?: ""
+      sequenceOf(Parser.Result(remaining, result))
+    } else {
+      emptySequence()
+    }
+  }
```

Comment:
Question : if the input only contains the delimiter (such as " " when `delimiter = ' '), should the parser parse an empty `Token` ?

If we want to avoid parsing empty tokens, a workaround could be to add a `filter { it.isNotEmpty() }` after the `Parser { }` definition.

Reply:
I think the most logical output it is an empty sequence, in this case. I will change it

Reply:
Fix in the commit https://github.com/epfl-SDP/android/pull/280/commits/dae24cc68d93bc15d8709b7af589eb1bd70b662d